core/api/core.api

```diff
@@ -6267,7 +6338,7 @@ public final class dev/kord/core/entity/GuildEmoji : dev/kord/core/entity/KordEn
 	public final fun getData ()Ldev/kord/core/cache/data/EmojiData;
 	public final fun getGuildId ()Ldev/kord/common/entity/Snowflake;
 	public fun getId ()Ldev/kord/common/entity/Snowflake;
-	public final fun getImage ()Ldev/kord/core/entity/Icon;
+	public final fun getImage ()Ldev/kord/core/entity/Asset;
```

Comment:
Yes that'a right: you can have two methods on jvm with same name and parameters when they have a different return type. In the bytecode they are referenced by their full signature - including return type.
So to preserve binary compatibility we have to keep the old method signature.
But since you can't have two methods (or properties which become functions on jvm) with the same name and parameters but a different return type in Kotlin, we have to use this pattern with `@JvmName` and `@Deprecated(..., level = HIDDEN)`.
If you add one of them and look at the resulting API dump, you'll see that the method will be there twice with the same name but a different return type - the only other difference being that the one marked as deprecated with `HIDDEN` level is synthetic, meaning it isn't visible to newly written source but is still available to already compiled bytecode which keeps the binary compatibility.

Reply:
Got it, thanks